// ntpac_api/src/model/CaptureListDTO.js
/**
 * NTPAC API
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 0.0.1
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.1-SNAPSHOT
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.NtpacApi) {
      root.NtpacApi = {};
    }
    root.NtpacApi.CaptureListDTO = factory(root.NtpacApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';




  /**
   * The CaptureListDTO model module.
   * @module model/CaptureListDTO
   * @version 0.0.1
   */

  /**
   * Constructs a new <code>CaptureListDTO</code>.
   * @alias module:model/CaptureListDTO
   * @class
   * @extends Object
   * @param analysisDuration {String} 
   * @param captureSize {Number} 
   * @param firstSeen {Date} 
   * @param id {String} 
   * @param l7ConversationCount {Number} 
   * @param lastSeen {Date} 
   * @param processed {Date} 
   */
  var exports = function(analysisDuration, captureSize, firstSeen, id, l7ConversationCount, lastSeen, processed) {
    var _this = this;

    _this['analysisDuration'] = analysisDuration;
    _this['captureSize'] = captureSize;
    _this['firstSeen'] = firstSeen;
    _this['id'] = id;
    _this['l7ConversationCount'] = l7ConversationCount;
    _this['lastSeen'] = lastSeen;
    _this['processed'] = processed;

    return _this;
  };

  /**
   * Constructs a <code>CaptureListDTO</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/CaptureListDTO} obj Optional instance to populate.
   * @return {module:model/CaptureListDTO} The populated <code>CaptureListDTO</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('analysisDuration')) {
        obj['analysisDuration'] = ApiClient.convertToType(data['analysisDuration'], 'String');
      }
      if (data.hasOwnProperty('captureSize')) {
        obj['captureSize'] = ApiClient.convertToType(data['captureSize'], 'Number');
      }
      if (data.hasOwnProperty('firstSeen')) {
        obj['firstSeen'] = ApiClient.convertToType(data['firstSeen'], 'Date');
      }
      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'String');
      }
      if (data.hasOwnProperty('l7ConversationCount')) {
        obj['l7ConversationCount'] = ApiClient.convertToType(data['l7ConversationCount'], 'Number');
      }
      if (data.hasOwnProperty('lastSeen')) {
        obj['lastSeen'] = ApiClient.convertToType(data['lastSeen'], 'Date');
      }
      if (data.hasOwnProperty('processed')) {
        obj['processed'] = ApiClient.convertToType(data['processed'], 'Date');
      }
      if (data.hasOwnProperty('uri')) {
        obj['uri'] = ApiClient.convertToType(data['uri'], 'String');
      }
    }
    return obj;
  }

  /**
   * @member {String} analysisDuration
   */
  exports.prototype['analysisDuration'] = undefined;
  /**
   * @member {Number} captureSize
   */
  exports.prototype['captureSize'] = undefined;
  /**
   * @member {Date} firstSeen
   */
  exports.prototype['firstSeen'] = undefined;
  /**
   * @member {String} id
   */
  exports.prototype['id'] = undefined;
  /**
   * @member {Number} l7ConversationCount
   */
  exports.prototype['l7ConversationCount'] = undefined;
  /**
   * @member {Date} lastSeen
   */
  exports.prototype['lastSeen'] = undefined;
  /**
   * @member {Date} processed
   */
  exports.prototype['processed'] = undefined;
  /**
   * @member {String} uri
   */
  exports.prototype['uri'] = undefined;



  return exports;
}));
