// ntpac_api/src/model/IPEndPointDTO.js
/**
 * NTPAC API
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 0.0.1
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.1-SNAPSHOT
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/AddressFamily'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./AddressFamily'));
  } else {
    // Browser globals (root is window)
    if (!root.NtpacApi) {
      root.NtpacApi = {};
    }
    root.NtpacApi.IPEndPointDTO = factory(root.NtpacApi.ApiClient, root.NtpacApi.AddressFamily);
  }
}(this, function(ApiClient, AddressFamily) {
  'use strict';




  /**
   * The IPEndPointDTO model module.
   * @module model/IPEndPointDTO
   * @version 0.0.1
   */

  /**
   * Constructs a new <code>IPEndPointDTO</code>.
   * @alias module:model/IPEndPointDTO
   * @class
   * @extends Object
   * @param addressFamily {module:model/AddressFamily} 
   * @param port {Number} 
   */
  var exports = function(addressFamily, port) {
    var _this = this;


    _this['addressFamily'] = addressFamily;
    _this['port'] = port;
    return _this;
  };

  /**
   * Constructs a <code>IPEndPointDTO</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/IPEndPointDTO} obj Optional instance to populate.
   * @return {module:model/IPEndPointDTO} The populated <code>IPEndPointDTO</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('address')) {
        obj['address'] = ApiClient.convertToType(data['address'], 'String');
      }
      if (data.hasOwnProperty('addressFamily')) {
        obj['addressFamily'] = AddressFamily.constructFromObject(data['addressFamily']);
      }
      if (data.hasOwnProperty('port')) {
        obj['port'] = ApiClient.convertToType(data['port'], 'Number');
      }
    }
    return obj;
  }

  /**
   * @member {String} address
   */
  exports.prototype['address'] = undefined;
  /**
   * @member {module:model/AddressFamily} addressFamily
   */
  exports.prototype['addressFamily'] = undefined;
  /**
   * @member {Number} port
   */
  exports.prototype['port'] = undefined;



  return exports;
}));
