// ntpac_api/src/model/L7ConversationListDTO.js
/**
 * NTPAC API
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 0.0.1
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.1-SNAPSHOT
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/IPEndPointDTO', 'model/IPProtocolType'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./IPEndPointDTO'), require('./IPProtocolType'));
  } else {
    // Browser globals (root is window)
    if (!root.NtpacApi) {
      root.NtpacApi = {};
    }
    root.NtpacApi.L7ConversationListDTO = factory(root.NtpacApi.ApiClient, root.NtpacApi.IPEndPointDTO, root.NtpacApi.IPProtocolType);
  }
}(this, function(ApiClient, IPEndPointDTO, IPProtocolType) {
  'use strict';




  /**
   * The L7ConversationListDTO model module.
   * @module model/L7ConversationListDTO
   * @version 0.0.1
   */

  /**
   * Constructs a new <code>L7ConversationListDTO</code>.
   * @alias module:model/L7ConversationListDTO
   * @class
   * @extends Object
   * @param firstSeen {Date} 
   * @param id {String} 
   * @param lastSeen {Date} 
   * @param pduCount {Number} 
   * @param protocolType {module:model/IPProtocolType} 
   */
  var exports = function(firstSeen, id, lastSeen, pduCount, protocolType) {
    var _this = this;


    _this['firstSeen'] = firstSeen;
    _this['id'] = id;
    _this['lastSeen'] = lastSeen;
    _this['pduCount'] = pduCount;
    _this['protocolType'] = protocolType;

    return _this;
  };

  /**
   * Constructs a <code>L7ConversationListDTO</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/L7ConversationListDTO} obj Optional instance to populate.
   * @return {module:model/L7ConversationListDTO} The populated <code>L7ConversationListDTO</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('destinationEndPoint')) {
        obj['destinationEndPoint'] = IPEndPointDTO.constructFromObject(data['destinationEndPoint']);
      }
      if (data.hasOwnProperty('firstSeen')) {
        obj['firstSeen'] = ApiClient.convertToType(data['firstSeen'], 'Date');
      }
      if (data.hasOwnProperty('id')) {
        obj['id'] = ApiClient.convertToType(data['id'], 'String');
      }
      if (data.hasOwnProperty('lastSeen')) {
        obj['lastSeen'] = ApiClient.convertToType(data['lastSeen'], 'Date');
      }
      if (data.hasOwnProperty('pduCount')) {
        obj['pduCount'] = ApiClient.convertToType(data['pduCount'], 'Number');
      }
      if (data.hasOwnProperty('protocolType')) {
        obj['protocolType'] = IPProtocolType.constructFromObject(data['protocolType']);
      }
      if (data.hasOwnProperty('sourceEndPoint')) {
        obj['sourceEndPoint'] = IPEndPointDTO.constructFromObject(data['sourceEndPoint']);
      }
    }
    return obj;
  }

  /**
   * @member {module:model/IPEndPointDTO} destinationEndPoint
   */
  exports.prototype['destinationEndPoint'] = undefined;
  /**
   * @member {Date} firstSeen
   */
  exports.prototype['firstSeen'] = undefined;
  /**
   * @member {String} id
   */
  exports.prototype['id'] = undefined;
  /**
   * @member {Date} lastSeen
   */
  exports.prototype['lastSeen'] = undefined;
  /**
   * @member {Number} pduCount
   */
  exports.prototype['pduCount'] = undefined;
  /**
   * @member {module:model/IPProtocolType} protocolType
   */
  exports.prototype['protocolType'] = undefined;
  /**
   * @member {module:model/IPEndPointDTO} sourceEndPoint
   */
  exports.prototype['sourceEndPoint'] = undefined;



  return exports;
}));
