// ntpac_api/src/model/L7PduDTO.js
/**
 * NTPAC API
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 0.0.1
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.1-SNAPSHOT
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/FlowDirection'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('./FlowDirection'));
  } else {
    // Browser globals (root is window)
    if (!root.NtpacApi) {
      root.NtpacApi = {};
    }
    root.NtpacApi.L7PduDTO = factory(root.NtpacApi.ApiClient, root.NtpacApi.FlowDirection);
  }
}(this, function(ApiClient, FlowDirection) {
  'use strict';




  /**
   * The L7PduDTO model module.
   * @module model/L7PduDTO
   * @version 0.0.1
   */

  /**
   * Constructs a new <code>L7PduDTO</code>.
   * @alias module:model/L7PduDTO
   * @class
   * @extends Object
   * @param direction {module:model/FlowDirection} 
   * @param firstSeenTicks {Number} 
   * @param lastSeenTicks {Number} 
   * @param firstSeen {Date} 
   * @param lastSeen {Date} 
   * @param payloadLen {Number} 
   */
  var exports = function(direction, firstSeenTicks, lastSeenTicks, firstSeen, lastSeen, payloadLen) {
    var _this = this;

    _this['direction'] = direction;
    _this['firstSeenTicks'] = firstSeenTicks;
    _this['lastSeenTicks'] = lastSeenTicks;
    _this['firstSeen'] = firstSeen;
    _this['lastSeen'] = lastSeen;

    _this['payloadLen'] = payloadLen;
    return _this;
  };

  /**
   * Constructs a <code>L7PduDTO</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/L7PduDTO} obj Optional instance to populate.
   * @return {module:model/L7PduDTO} The populated <code>L7PduDTO</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) {
      obj = obj || new exports();

      if (data.hasOwnProperty('direction')) {
        obj['direction'] = FlowDirection.constructFromObject(data['direction']);
      }
      if (data.hasOwnProperty('firstSeenTicks')) {
        obj['firstSeenTicks'] = ApiClient.convertToType(data['firstSeenTicks'], 'Number');
      }
      if (data.hasOwnProperty('lastSeenTicks')) {
        obj['lastSeenTicks'] = ApiClient.convertToType(data['lastSeenTicks'], 'Number');
      }
      if (data.hasOwnProperty('firstSeen')) {
        obj['firstSeen'] = ApiClient.convertToType(data['firstSeen'], 'Date');
      }
      if (data.hasOwnProperty('lastSeen')) {
        obj['lastSeen'] = ApiClient.convertToType(data['lastSeen'], 'Date');
      }
      if (data.hasOwnProperty('payload')) {
        obj['payload'] = ApiClient.convertToType(data['payload'], 'Blob');
      }
      if (data.hasOwnProperty('payloadLen')) {
        obj['payloadLen'] = ApiClient.convertToType(data['payloadLen'], 'Number');
      }
    }
    return obj;
  }

  /**
   * @member {module:model/FlowDirection} direction
   */
  exports.prototype['direction'] = undefined;
  /**
   * @member {Number} firstSeenTicks
   */
  exports.prototype['firstSeenTicks'] = undefined;
  /**
   * @member {Number} lastSeenTicks
   */
  exports.prototype['lastSeenTicks'] = undefined;
  /**
   * @member {Date} firstSeen
   */
  exports.prototype['firstSeen'] = undefined;
  /**
   * @member {Date} lastSeen
   */
  exports.prototype['lastSeen'] = undefined;
  /**
   * @member {Blob} payload
   */
  exports.prototype['payload'] = undefined;
  /**
   * @member {Number} payloadLen
   */
  exports.prototype['payloadLen'] = undefined;



  return exports;
}));
