// ntpac_api/src/index.js
/**
 * NTPAC API
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 0.0.1
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.1-SNAPSHOT
 *
 * Do not edit the class manually.
 *
 */

(function(factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['./ApiClient', 'model/AddressFamily', 'model/CaptureDetailDTO', 'model/CaptureListDTO', 'model/FlowDirection', 'model/IPEndPointDTO', 'model/IPProtocolType', 'model/L7ConversationDetailDTO', 'model/L7ConversationListDTO', 'model/L7PduDTO', 'api/CaptureApi', 'api/L7ConversationApi'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('./ApiClient'), require('./model/AddressFamily'), require('./model/CaptureDetailDTO'), require('./model/CaptureListDTO'), require('./model/FlowDirection'), require('./model/IPEndPointDTO'), require('./model/IPProtocolType'), require('./model/L7ConversationDetailDTO'), require('./model/L7ConversationListDTO'), require('./model/L7PduDTO'), require('./api/CaptureApi'), require('./api/L7ConversationApi'));
  }
}(function(ApiClient, AddressFamily, CaptureDetailDTO, CaptureListDTO, FlowDirection, IPEndPointDTO, IPProtocolType, L7ConversationDetailDTO, L7ConversationListDTO, L7PduDTO, CaptureApi, L7ConversationApi) {
  'use strict';

  /**
   * ERROR_UNKNOWN.<br>
   * The <code>index</code> module provides access to constructors for all the classes which comprise the public API.
   * <p>
   * An AMD (recommended!) or CommonJS application will generally do something equivalent to the following:
   * <pre>
   * var NtpacApi = require('index'); // See note below*.
   * var xxxSvc = new NtpacApi.XxxApi(); // Allocate the API class we're going to use.
   * var yyyModel = new NtpacApi.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * <em>*NOTE: For a top-level AMD script, use require(['index'], function(){...})
   * and put the application logic within the callback function.</em>
   * </p>
   * <p>
   * A non-AMD browser application (discouraged) might do something like this:
   * <pre>
   * var xxxSvc = new NtpacApi.XxxApi(); // Allocate the API class we're going to use.
   * var yyy = new NtpacApi.Yyy(); // Construct a model instance.
   * yyyModel.someProperty = 'someValue';
   * ...
   * var zzz = xxxSvc.doSomething(yyyModel); // Invoke the service.
   * ...
   * </pre>
   * </p>
   * @module index
   * @version 0.0.1
   */
  var exports = {
    /**
     * The ApiClient constructor.
     * @property {module:ApiClient}
     */
    ApiClient: ApiClient,
    /**
     * The AddressFamily model constructor.
     * @property {module:model/AddressFamily}
     */
    AddressFamily: AddressFamily,
    /**
     * The CaptureDetailDTO model constructor.
     * @property {module:model/CaptureDetailDTO}
     */
    CaptureDetailDTO: CaptureDetailDTO,
    /**
     * The CaptureListDTO model constructor.
     * @property {module:model/CaptureListDTO}
     */
    CaptureListDTO: CaptureListDTO,
    /**
     * The FlowDirection model constructor.
     * @property {module:model/FlowDirection}
     */
    FlowDirection: FlowDirection,
    /**
     * The IPEndPointDTO model constructor.
     * @property {module:model/IPEndPointDTO}
     */
    IPEndPointDTO: IPEndPointDTO,
    /**
     * The IPProtocolType model constructor.
     * @property {module:model/IPProtocolType}
     */
    IPProtocolType: IPProtocolType,
    /**
     * The L7ConversationDetailDTO model constructor.
     * @property {module:model/L7ConversationDetailDTO}
     */
    L7ConversationDetailDTO: L7ConversationDetailDTO,
    /**
     * The L7ConversationListDTO model constructor.
     * @property {module:model/L7ConversationListDTO}
     */
    L7ConversationListDTO: L7ConversationListDTO,
    /**
     * The L7PduDTO model constructor.
     * @property {module:model/L7PduDTO}
     */
    L7PduDTO: L7PduDTO,
    /**
     * The CaptureApi service constructor.
     * @property {module:api/CaptureApi}
     */
    CaptureApi: CaptureApi,
    /**
     * The L7ConversationApi service constructor.
     * @property {module:api/L7ConversationApi}
     */
    L7ConversationApi: L7ConversationApi
  };

  return exports;
}));
