// ntpac_api/src/api/L7ConversationApi.js
/**
 * NTPAC API
 * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
 *
 * OpenAPI spec version: 0.0.1
 *
 * NOTE: This class is auto generated by the swagger code generator program.
 * https://github.com/swagger-api/swagger-codegen.git
 *
 * Swagger Codegen version: 2.4.1-SNAPSHOT
 *
 * Do not edit the class manually.
 *
 */

(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['ApiClient', 'model/L7ConversationDetailDTO', 'model/L7ConversationListDTO'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'), require('../model/L7ConversationDetailDTO'), require('../model/L7ConversationListDTO'));
  } else {
    // Browser globals (root is window)
    if (!root.NtpacApi) {
      root.NtpacApi = {};
    }
    root.NtpacApi.L7ConversationApi = factory(root.NtpacApi.ApiClient, root.NtpacApi.L7ConversationDetailDTO, root.NtpacApi.L7ConversationListDTO);
  }
}(this, function(ApiClient, L7ConversationDetailDTO, L7ConversationListDTO) {
  'use strict';

  /**
   * L7Conversation service.
   * @module api/L7ConversationApi
   * @version 0.0.1
   */

  /**
   * Constructs a new L7ConversationApi. 
   * @alias module:api/L7ConversationApi
   * @class
   * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
   * default to {@link module:ApiClient#instance} if unspecified.
   */
  var exports = function(apiClient) {
    this.apiClient = apiClient || ApiClient.instance;


    /**
     * Callback function to receive the result of the l7ConversationDelete operation.
     * @callback module:api/L7ConversationApi~l7ConversationDeleteCallback
     * @param {String} error Error message, if any.
     * @param data This operation does not return a value.
     * @param {String} response The complete HTTP response.
     */

    /**
     * @param {String} id 
     * @param {module:api/L7ConversationApi~l7ConversationDeleteCallback} callback The callback function, accepting three arguments: error, data, response
     */
    this.l7ConversationDelete = function(id, callback) {
      var postBody = null;

      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling l7ConversationDelete");
      }


      var pathParams = {
        'id': id
      };
      var queryParams = {
      };
      var collectionQueryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = null;

      return this.apiClient.callApi(
        '/api/L7Conversation/{id}', 'DELETE',
        pathParams, queryParams, collectionQueryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the l7ConversationGet operation.
     * @callback module:api/L7ConversationApi~l7ConversationGetCallback
     * @param {String} error Error message, if any.
     * @param {module:model/L7ConversationDetailDTO} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * @param {String} id 
     * @param {module:api/L7ConversationApi~l7ConversationGetCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/L7ConversationDetailDTO}
     */
    this.l7ConversationGet = function(id, callback) {
      var postBody = null;

      // verify the required parameter 'id' is set
      if (id === undefined || id === null) {
        throw new Error("Missing the required parameter 'id' when calling l7ConversationGet");
      }


      var pathParams = {
        'id': id
      };
      var queryParams = {
      };
      var collectionQueryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = L7ConversationDetailDTO;

      return this.apiClient.callApi(
        '/api/L7Conversation/{id}', 'GET',
        pathParams, queryParams, collectionQueryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the l7ConversationGetAll operation.
     * @callback module:api/L7ConversationApi~l7ConversationGetAllCallback
     * @param {String} error Error message, if any.
     * @param {Array.<module:model/L7ConversationListDTO>} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * @param {module:api/L7ConversationApi~l7ConversationGetAllCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link Array.<module:model/L7ConversationListDTO>}
     */
    this.l7ConversationGetAll = function(callback) {
      var postBody = null;


      var pathParams = {
      };
      var queryParams = {
      };
      var collectionQueryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = [L7ConversationListDTO];

      return this.apiClient.callApi(
        '/api/L7Conversation', 'GET',
        pathParams, queryParams, collectionQueryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }

    /**
     * Callback function to receive the result of the l7ConversationGetByCapture operation.
     * @callback module:api/L7ConversationApi~l7ConversationGetByCaptureCallback
     * @param {String} error Error message, if any.
     * @param {Array.<module:model/L7ConversationListDTO>} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * @param {String} captureId 
     * @param {module:api/L7ConversationApi~l7ConversationGetByCaptureCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link Array.<module:model/L7ConversationListDTO>}
     */
    this.l7ConversationGetByCapture = function(captureId, callback) {
      var postBody = null;

      // verify the required parameter 'captureId' is set
      if (captureId === undefined || captureId === null) {
        throw new Error("Missing the required parameter 'captureId' when calling l7ConversationGetByCapture");
      }


      var pathParams = {
        'captureId': captureId
      };
      var queryParams = {
      };
      var collectionQueryParams = {
      };
      var headerParams = {
      };
      var formParams = {
      };

      var authNames = [];
      var contentTypes = ['application/json'];
      var accepts = ['application/json'];
      var returnType = [L7ConversationListDTO];

      return this.apiClient.callApi(
        '/api/L7Conversation/capture/{captureId}', 'GET',
        pathParams, queryParams, collectionQueryParams, headerParams, formParams, postBody,
        authNames, contentTypes, accepts, returnType, callback
      );
    }
  };

  return exports;
}));
